scripts/compare.py

```diff
@@ -0,0 +1,232 @@
+				clean_path = append_svg_polygons(xml, parent, [{"%s[%d]" % (polygon_id, i): polygon}], style)
+				assert(len(clean_path) == 1)
+				clean_paths.append(clean_path[0])
+
+			result.append(clean_paths)
+
+			continue
+
+		path = SvgPath()
+		assert(len(polygon.interiors) == 0)
+		for coord in polygon.exterior.coords:
+			if len(path) == 0:
+				path.append(SvgMove(complex(*coord)))
+			else:
+				path.append(SvgLine(path[-1].end, complex(*coord)))
+
+		clean_path = xml.createElement("path")
+		clean_path.setAttribute('id', polygon_id)
+		clean_path.setAttribute('d', path.d())
+		clean_path.setAttribute('style', style)
+		parent.appendChild(clean_path)
+
+		result.append([clean_path])
+
+	return result
+
+def find_group(xml, label):
+	groups = xml.getElementsByTagName('g')
+	for group in groups:
+		if group.getAttribute('inkscape:label') == label:
+			print("Found group '%s' (id: '%s')" % (label, group.getAttribute('id')))
+			return group
+	return None
+
+def compare(old_polygons, new_polygons):
+
+	duplicate = {} # Exact matches
+	touching = {} # Small overlap
+	conflict = {} # Large overlap
+	added = [] # No overlap, exists only in new
+
+	# Compare
+	for new_element_id, new_polygon in new_polygons.items():
+		print('Processing SVG child elements:', new_element_id)
+
+		overlaps = {}
+
+		polygon = new_polygon
+		for old_element_id, old_polygon in old_polygons.items():
+
+			if not new_polygon.intersects(old_polygon):
+				#FIXME: Get distance to check if someone might have left an accidental gap, or moved too close
+				pass
+			else:
+				# Reject, if this is the exact same shape
+				if new_polygon.equals(old_polygon): #FIXME: almost_equals?
+					print("warning: %s is a duplicate (equals existing %s)" % (new_element_id, old_element_id))
+					duplicate[new_element_id] = old_element_id
+					continue
+
+				# Get overlap
+				overlap_polygons = new_polygon.intersection(old_polygon)
+
+				# Ensure this is a multipolygon
+				if not isinstance(overlap_polygons, shapely.geometry.MultiPolygon):
+					overlap_polygons = shapely.geometry.MultiPolygon([overlap_polygons])
+
+				# Check each overlap separately
+				for overlap_polygon in overlap_polygons:
+					export_overlap = False
+
+					if overlap_polygon.area > 0:
+						if overlap_polygon.area <= (stroke_width*stroke_width * 2.0):
+							#FIXME: Check if this is in a pad or via; if not in pad or via: error
+							print("note: Small overlap, assuming connection")
+							touching[new_element_id] = old_element_id
+						else:
+							print("error: Large overlap")
+							conflict[new_element_id] = old_element_id
+							export_overlap = True
+
+					if export_overlap:
+						overlaps[old_element_id] = overlaps.get(old_element_id, []).append(overlap_polygon)
+
+		# Check if this was newly added to keep track of it
+		if new_element_id not in conflict:
+			if new_element_id not in duplicate:
+				added.append(new_element_id)
+
+		#FIXME: This exports the path and its overlap
```

Comment:
I believe the FIXME is resolved. I'm not sure what I meant.
I assume this was a placeholder - but obviously it's implemented now.